src/lib/broadcast.jl

```diff
@@ -272,7 +272,8 @@ end
 
   end
 
-  @adjoint (::Type{T})(xs::Array) where {T <: CUDA.CuArray} =
+  _eyelike(y::CUDA.CuVector{T}) where T = CUDA.CuArray(I(length(y)))
+  @adjoint (::Type{T})(xs::AbstractArray) where {T <: CUDA.CuArray} =
```

Comment:
IIUC the intent of the PR, would we not want a dedicated path `UniformScaling`? Taking `<:AbstractArray` presumes we can convert back to that subtype, but the implementation can only convert back to `Array` at present. Adapt.jl may be able to help us here.